src/python/pants/notes/2.5.x.md

```diff
@@ -0,0 +1,37 @@
+# 2.5.x Stable Releases
+
+See https://www.pantsbuild.org/v2.5/docs/release-notes-2-5 for an overview of the changes in this release series.
+
+## 2.5.0.dev0 (Apr 02, 2021)
+
+### New Features
+
+* Subset lockfile resolves. ([#11773](https://github.com/pantsbuild/pants/pull/11773))
+
+* Upgrade Pex to 2.1.38. ([#11829](https://github.com/pantsbuild/pants/pull/11829))
```

Comment:
Internal change. It was technically a bug fix, but the bug never impacted Pants users and it was prework for "Subset lockfile resolves"

Reply:
It is a public change - I think plugin API is the closest appropriate bucket. Since plugins depend on everything pantsbuild.pants depends on, when we change a pin we change our rules API and potentially break rule writers who happen to also depend on an intersction of project names but not versions.

Reply:
I've moved it to the plugin API section.

Note that this bumped from 2.1.35 to 2.1.38, so encompassed a few versions :)